app/lib/search_query_transformer.rb

```diff
@@ -224,6 +224,17 @@ def language_code_from_term(term)
     end
   end
 
+  class TermValidator
+    STRICT_DATE_REGEX = /\A\d{4}-\d{2}-\d{2}\z/ # yyyy-MM-dd
+    EPOCH_MILLIS_REGEX = /\A\d{1,11}\z/
```

Comment:
Possibly not a matter now, but could this be impacted with 64 bit epochs vs 32 bit ones?

Reply:
sounds good, updated to support max values of `epoch_millis` which ES docs say is Java's `Long.MAX_VALUE` 2^63. it seems like it should technically support `Long.MIN_VALUE` as well, but I wasn't able to get those queries to work on my local ES